src/nnet3/nnet-utils.cc

```diff
@@ -326,7 +326,7 @@ void ScaleNnetComponents(const Vector<BaseFloat> &scale_factors,
         KALDI_ERR << "Updatable component does not inherit from class "
             "UpdatableComponent; change this code.";
       KALDI_ASSERT(i < scale_factors.Dim());
-      uc->Scale(scale_factors(i));
+      uc->Scale(scale_factors(i++));
     }
   }
```

Comment:
assert at the end that i == scale_factors.Dim().  You'd have to declare i earlier.  The same in SetLearningRates.